sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1460,6 +1460,9 @@ object GpuOverrides {
           GpuPythonUDF(a.name, a.func, a.dataType,
             childExprs.map(_.convertToGpu()),
             a.evalType, a.udfDeterministic, a.resultId)
+
+        override def isSupportedType(t: DataType): Boolean =
+          GpuOverrides.isSupportedType(t, allowArray = true, allowNesting = true)
```

Comment:
We have not tested array inputs or nested arrays.  I would like to see some basic tests for those too, or else we need to add some checks so that we don't convert plans with them in it if we cannot support them yet.

Reply:
Since there is something wrong in writing array data by cudf arrow IPC writer, i chose to add some checks.
I have tried excluding all the possible cases which are enabled but not intended. If any missing, i will take it as a follow.